### DISCUSSION TITLE
When to use Native database type attribute of MongoDB？

### DISCUSSION THREAD

**Discussion by zwl1619** [prisma/prisma#21915] [Q&A]
### Question

There are four Native database type attribute of MongoDB in the Prisma docs here:
https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference
```
@db.String
@db.ObjectId  //Required if the underlying BSON type is OBJECT_ID (ID fields, relation scalars)
@db.Int
@db.Long
```

"@db.ObjectId" has an annotation for use, the other ones don't have annotations. 
So when to use "@db.String",  "@db.Int", "@db.Long"?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @zwl1619 👋 

Thank you for raising this question.

The `@db` attribute in Prisma is used to define the exact database type in the Prisma schema. This is particularly useful when you want to leverage database-specific features.

In the context of MongoDB, the `@db` attribute can be used with different scalar types like `@db.String`, `@db.Int`, `@db.Long`, and `@db.ObjectId`. Here's when you might use each:

`@db.String`: This attribute is used when you want the corresponding MongoDB type to be a string. In Prisma, the String scalar type maps to string in MongoDB.

`@db.Int`: This attribute is used when you want the corresponding MongoDB type to be an integer. In Prisma, the Int scalar type maps to int in MongoDB.

`@db.Long`: This attribute is used when you want the corresponding MongoDB type to be a long integer. In Prisma, the BigInt scalar type maps to long in MongoDB.

`@db.ObjectId`: This attribute is used when you want the corresponding MongoDB type to be an `ObjectId`. This is typically used for defining `ID` fields and relation scalar fields in your Prisma schema. For example, if your model's `ID` is an `ObjectId` (represented by a String field), you must add `@db.ObjectId` to the model's `ID` and the relation scalar field on the other side of the relation.

Here's an example of how you might use these attributes in your Prisma schema:
```prisma
model User {
  id String @id @default(auto()) @map("_id") @db.ObjectId
  name String @db.String
  age Int @db.Int
  largeNumber BigInt @db.Long
}
```
In this example, the `id` field is an `ObjectId`, the `name` field is a string, the `age` field is an integer, and the `largeNumber` field is a long integer.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
